sql/analyzer/hoist_select_exists.go

```diff
@@ -159,12 +196,75 @@ func decorrelateOuterCols(e *plan.Subquery, scopeLen int) *hoistSubquery {
 	})
 
 	if same || len(outerFilters) == 0 {
-		return nil
+		return nil, nil
+	}
+
+	nodeAliases, err := getTableAliases(n, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	conflicts, nonConflicted := outsideAliases.findConflicts(nodeAliases)
+	for _, goodAlias := range nonConflicted {
+		target, ok := nodeAliases[goodAlias]
+		if !ok {
+			return nil, fmt.Errorf("node alias %s is not in nodeAliases", goodAlias)
```

Comment:
All of these errors are potentially returned to end users. I'm not sure I would know understand or know what to do with this message. This is pretty far outside of what should ever be possible, also. Analysis is always going to be single threaded.

Reply:
This should not happen, but it could if there's a bug. Should we panic instead? Try to call `add` even if `table` is `nil`?

Reply:
idk, i might just do `target, _ := nodeAliases[goodAlias]`. Would probably be fine to leave also. Just seems strange to read a list from a hash map and then be paranoid the same values exited the map in a single thread context where we only ever append to the map.

Reply:
Going to leave as-is. It's a very specific case and should only fail if there's a serious bug. Users shouldn't be seeing this otherwise.